### ISSUE TITLE
Systemd socket activation

### ISSUE THREAD

**Issue by fxposter** [bflad/chef-docker#278]
Reading this question and answers -  http://stackoverflow.com/questions/24791648/start-docker-container-using-systemd-socket-activation - I wonder why do we have `.socket` files created for systemd services, as long as docker cannot use them?
And, actually, I'm getting errors when starting services, cause systemd complains that it can't start container, cause the port is already occupied.

**Comment by bflad** [comment]
Probably us being over-proactive. Would you mind pasting one of the errors here? We can certainly disable socket activation for containers if its breaking things.

**Comment by marc-** [comment]
I get following error for CentOS 7.0.1406:
```
       [2015-04-21T12:13:44+00:00] ERROR: service[docker] (docker::systemd line 25) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '6'                                                     
       ---- Begin output of /bin/systemctl restart docker ----
       STDOUT: 
       STDERR: Failed to issue method call: Unit docker.socket failed to load: Invalid argument. See system logs and 'systemctl status docker.socket' for details.                                                                              
       ---- End output of /bin/systemctl restart docker ----
       Ran /bin/systemctl restart docker returned 6
       [2015-04-21T12:13:44+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```
My `docker.socker` file:
```
[Unit]
Description=Docker Socket for the API
PartOf=docker.service

[Socket]
ListenStream=
SocketMode=0660
SocketUser=root
SocketGroup=root

[Install]
WantedBy=sockets.target
```
Output of the `systemctl status -l docker.socket`
```
docker.socket - Docker Socket for the API
   Loaded: error (Reason: Invalid argument)
   Active: inactive (dead)

апр 21 12:13:44 client-centos-70 systemd[1]: [/usr/lib/systemd/system/docker.socket:8] Unknown lvalue 'SocketUser' in section 'Socket'
апр 21 12:13:44 client-centos-70 systemd[1]: [/usr/lib/systemd/system/docker.socket:9] Unknown lvalue 'SocketGroup' in section 'Socket'
апр 21 12:13:44 client-centos-70 systemd[1]: docker.socket lacks Listen setting. Refusing.
```

But I think that is more about `docker.socker.erb` [template](https://github.com/bflad/chef-docker/blob/master/templates/default/docker.socket.erb#L7) and [attributes](https://github.com/bflad/chef-docker/blob/master/attributes/default.rb#L153) . If I put `/var/run/docker.sock` as value for `ListenStream` docker service starts.

**Comment by miguelaferreira** [comment]
I'm seeing the same error and the solution prosed by @marc- fixed it.

**Comment by jslusher** [comment]
I'm getting this as well. @marc- is right about the fix. Is this a bug or is it something I'm overlooking in the configuration?

**Comment by jslusher** [comment]
It appears the fix is in the master branch for version 0.37.0, but not in the tagged 0.37.0 that you get by default from the chef supermarket. I had to add ", branch: 'master'" for the cookbook to my Berksfile to get it to pull down the code that accounts for this problem.

**Comment by someara** [comment]
It looks like this was a mis-feature in the previous versions.

Systemd socket activation is basically "run docker from xinetd".

This could make for an interesting service provider in the post-1.0.0 future, but I'm gonna close this for now.

-s
